core/migration/runner.go

```diff
@@ -0,0 +1,71 @@
+package migration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+	"github.com/anyproto/anytype-heart/pkg/lib/logging"
+	"github.com/anyproto/anytype-heart/space/clientspace"
+)
+
+const (
+	loggerName = "migration-runner"
+	errFormat  = "failed to run migration '%s' in space '%s': %w. %d out of %d objects were migrated"
+)
+
+var log = logging.Logger(loggerName)
+
+type Migration interface {
+	Run(context.Context, QueryableStore, DoableViaContext) (toMigrate, migrated int, err error)
+	Name() string
+}
+
+func Run(ctx context.Context, store objectstore.ObjectStore, space clientspace.Space) {
+	if err := run(ctx, store, space,
+		systemObjectReviser{},
+		readonlyRelationsFixer{},
+	); err != nil {
+		log.Errorf("failed to run default migrations: %v", err)
+	}
+}
+
+func run(ctx context.Context, store objectstore.ObjectStore, space clientspace.Space, migrations ...Migration) (mErr error) {
+	var (
+		spaceId = space.Id()
+		finish  = make(chan struct{})
+		sstore  = &safeStore{store: store}
```

Comment:
Maybe rename sstore to something more meaningful?

Reply:
Done!